### ISSUE TITLE
[Bug]: Code Freeze date for Core and Dashboard should be atleast 2 business days before code freeze for Plugins

### ISSUE THREAD

**Issue by minalsha** [opensearch-project/opensearch-build#3924]
### Describe the bug

As per https://github.com/opensearch-project/opensearch-build/issues/3743, for 2.10.0, the code freeze date(code complete, release branch and version increments) for Core and Dashboard is set to 9/4 and for respective plugins to be part of 2.10.0 as well, it is same as seen: https://github.com/opensearch-project/job-scheduler/issues/424. The first RC cut is called out 9/5 as per https://opensearch.org/releases.html which means that all plugins alongwith core and dashboard need to be code, test complete, release branch and version increments by 9/4. 

My concern is that, plugins are downstream of Core and Dashboard. So if there are any last minute changes to either Core/Dashboard on 9/4, downstream plugins which may get impacted would have 0 to less time to address the changes/fixes due to upstream changes. 

**My request here is to have Code freeze/release readiness date for both Core and Dashboard to be atleast 2 days before code freeze/release readiness for plugins.** 

### To reproduce

N/A

**Comment by minalsha** [comment]
cc @CEHENKLE , @bbarani , @dblock , @navneet1v

**Comment by dblock** [comment]
This was hotly debated multiple times. Would this apply to all dependencies (e.g. many plugins depend on job scheduler)? Would an alternative of adding "feature complete", after which "breaking changes" aren't allowed, give us more flexibility and time?

**Comment by minalsha** [comment]
> This was hotly debated multiple times. Would this apply to all dependencies (e.g. many plugins depend on job scheduler)? Would an alternative of adding "feature complete", after which "breaking changes" aren't allowed, give us more flexibility and time?

Yes this does apply to JS plugin as well. My concern is if we know there is a dependency chain, we want to ensure whateve is the upstream, should be ready before downstream.

**Comment by gaiksaya** [comment]
@bbarani Is https://github.com/opensearch-project/.github/issues the right place for this discussion?
